source/common/redis/command_splitter_impl.cc

```diff
@@ -38,19 +53,24 @@ SplitRequestPtr SimpleRequest::create(ConnPool::Instance& conn_pool,
   return std::move(request_ptr);
 }
 
-void SimpleRequest::onResponse(RespValuePtr&& response) {
-  handle_ = nullptr;
-  callbacks_.onResponse(std::move(response));
-}
+SplitRequestPtr EvalRequest::create(ConnPool::Instance& conn_pool,
+                                    const RespValue& incoming_request, SplitCallbacks& callbacks) {
 
-void SimpleRequest::onFailure() {
-  handle_ = nullptr;
-  callbacks_.onResponse(Utility::makeError("upstream failure"));
-}
+  if (incoming_request.asArray().size() < 4) {
+    callbacks.onResponse(Utility::makeError("wrong number of arguments for command"));
+    return nullptr;
+  }
 
-void SimpleRequest::cancel() {
-  handle_->cancel();
-  handle_ = nullptr;
+  std::unique_ptr<EvalRequest> request_ptr{new EvalRequest(callbacks)};
+
+  request_ptr->handle_ = conn_pool.makeRequest(incoming_request.asArray()[3].asString(),
```

Comment:
I don't really know much about the Redis command format, so wondering why the magic 4 here. Could you add a one-line comment with the example array structure for a command that might illustrate?